libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -1037,6 +1043,17 @@ void NavEKF3_core::CovariancePrediction()
         P[21][21] = P[19][19];
     }
 
+    if (needEarthBodyVarReset) {
+        // reset mag earth field variances
+        needEarthBodyVarReset = false;
+        zeroCols(P,16,18);
+        zeroRows(P,16,18);
+        P[16][16] = sq(frontend->_magNoise);
+        P[17][17] = P[16][16];
+        P[18][18] = P[16][16];
+        FuseDeclination(0.34f);
```

Comment:
why 0.34 radians error?

Reply:
20 degrees.

Reply:
It's sufficient to give the earth field some yaw stability.